drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -83,16 +117,24 @@ local function emit_light_status_events(light_device, light)
       local kelvin = math.floor(
         st_utils.clamp_value(handlers.mirek_to_kelvin(mirek), min, HueApi.MAX_TEMP_KELVIN)
       )
-      light_device:emit_event(capabilities.colorTemperature.colorTemperature(kelvin))
+      if last_emitted.color_temperature ~= kelvin then
+        light_device:emit_event(capabilities.colorTemperature.colorTemperature(kelvin))
+        last_emitted.color_temperature = kelvin
+      end
     end
 
     if light.color then
       light_device:set_field(Fields.GAMUT, light.color.gamut, { persist = true })
-      local r, g, b = HueColorUtils.safe_xy_to_rgb(light.color.xy, light.color.gamut)
-      local hue, sat, _ = st_utils.rgb_to_hsv(r, g, b)
-
-      light_device:emit_event(capabilities.colorControl.hue(st_utils.round(hue * 100)))
-      light_device:emit_event(capabilities.colorControl.saturation(st_utils.round(sat * 100)))
+      if not _xy_nearly_equal(light.color.xy, last_emitted.color.xy) then
+        local r, g, b = HueColorUtils.safe_xy_to_rgb(light.color.xy, light.color.gamut)
+        local hue, sat, _ = st_utils.rgb_to_hsv(r, g, b)
+
+        local hue_emit = st_utils.round(hue * 100)
+        local sat_emit = st_utils.round(sat * 100)
+        light_device:emit_event(capabilities.colorControl.hue(hue_emit))
+        light_device:emit_event(capabilities.colorControl.saturation(sat_emit))
+        last_emitted.color.xy = light.color.xy
```

Comment:
Why are you setting this, won't it go out of scope at the end of this function? Should you be setting a field?